modules/drivers/vertica/src/metabase/driver/vertica.clj

```diff
@@ -75,7 +75,7 @@
     expr))
 
 (defn- date-trunc [unit expr] (hsql/call :date_trunc (hx/literal unit) (cast-timestamp expr)))
-(defn- extract    [unit expr] (hsql/call :extract    unit              expr))
+(defn- extract    [unit expr] (hsql/call :extract    unit              (cast-timestamp expr)))
```

Comment:
Having to change `extract` to get vertica to work here suggests to me that datetime literals aren't being spliced into the SQL query correctly for vertica.

Does vertica need implementations for `metabase.driver.sql.util.unprepare/unprepare-value`? e.g.
https://github.com/metabase/metabase/blob/b2791cf8d5d8022ac9d4499332c5ea2f5cceb3be/modules/drivers/snowflake/src/metabase/driver/snowflake.clj#L346

Reply:
unprepare is actually irrelevant here.

unprepare is used for the `convert to SQL` feature where we convert a native with params, to native without params.

Reply:
Right. It still seems strange that vertica's datetime literals aren't being spliced with the same effect as `unprepare-value`. But I suppose it's an improvement given we're already doing it for `date-trunc`.